polaris-react/src/components/ActionList/components/Item/Item.tsx

```diff
@@ -61,17 +62,27 @@ export function Item({
     );
   }
 
+  const truncate = !ellipsis && contentOverflow === 'truncate';
```

Comment:
Not required if ellipsis is removed: https://github.com/Shopify/polaris/pull/9005/files